### PR TITLE
Feat/indikator 10

### PR DESCRIPTION
**Pull request by afifahfq**
Text finding for indicator 10
Notes:
- checked for keywords in lvl 2 and lvl 4
- might remove lvl 5 if not needed

Example result for Batu Bara:
image

### CODE REVIEW

File: textfinding_indikator10.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    # read line by line from txt
+    for line in file:
+
+        if (lv == 1):
+            reg = f'(?:(tim)\s+(koordinasi)\s+(spbe)?)'
+            if (re.search(reg, line, re.IGNORECASE)):
+                result.append([idx, line])
+
+        elif (lv == 2):
+            for key in list(keyword):
```
Comment (AlifahRBasyasya) [inline textfinding_indikator10.py:28]: Same as [#3 (comment)](https://github.com/Tugas-Akhir-SPBE-IF-2019/ta-spbe-textfinding/pull/3#discussion_r1140977512)

File: textfinding_indikator10.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                if key in keyword:
+                    if re.search(key, line, re.IGNORECASE):
+
+                        # include words
+                        reg = f'(?:(tim)\s+(koordinasi)\s+(spbe)?)'
+                        if re.search(reg, line, re.IGNORECASE):
+
+                            # exclude words
+                            if (not excludewords(line)):
```
Comment (AlifahRBasyasya) [inline textfinding_indikator10.py:37]: Same as [#5 (comment)](https://github.com/Tugas-Akhir-SPBE-IF-2019/ta-spbe-textfinding/pull/5#discussion_r1142378703)

File: textfinding_indikator10.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                                result.append([idx, line])
+                                keyword.remove(key)
+
+        elif (lv == 4):
+            reg = f'(?:(tim)\s+(koordinasi))'
+            if re.search(reg, line, re.IGNORECASE):
+                res = [ele for ele in keyword if (ele in line)]
+                if (res):
+                    if (not excludewords(line)):
```
Comment (AlifahRBasyasya) [inline textfinding_indikator10.py:46]: Same as [#5 (comment)](https://github.com/Tugas-Akhir-SPBE-IF-2019/ta-spbe-textfinding/pull/5#discussion_r1142378703)

File: textfinding_indikator10.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                    if re.search(key, line, re.IGNORECASE):
+
+                        # include words
+                        reg = f'(?:(tim)\s+(koordinasi)\s+(spbe)?)'
+                        if re.search(reg, line, re.IGNORECASE):
+
+                            # exclude words
+                            if (not excludewords(line)):
+                                result.append([idx, line])
+                                keyword.remove(key)
```
Comment (AlifahRBasyasya) [inline textfinding_indikator10.py:39]: Same as [#3 (comment)](https://github.com/Tugas-Akhir-SPBE-IF-2019/ta-spbe-textfinding/pull/3#discussion_r1140977512)

File: textfinding_indikator10.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                if (res):
+                    if (not excludewords(line)):
+                        result.append([idx, line])
+
+        else:  # lvl 5
+            res = [ele for ele in keyword if (ele in line)]
+            if (res):
+                reg = f'(?:(tim)\s+(koordinasi)\s+(spbe)?)'
+                if re.search(reg, line, re.IGNORECASE):
+                    if (not excludewords(line)):
```
Comment (AlifahRBasyasya) [inline textfinding_indikator10.py:54]: Same as [#5 (comment)](https://github.com/Tugas-Akhir-SPBE-IF-2019/ta-spbe-textfinding/pull/5#discussion_r1142378190)

File: textfinding_indikator10.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        idx += 1
+
+    file.close()
+    return (result)
+
+
+# if __name__ == '__main__':
+#     pdfparser(sys.argv[1])
+
+def excludewords(line):
```
Comment (AlifahRBasyasya) [inline textfinding_indikator10.py:66]: Same as [#5 (comment)](https://github.com/Tugas-Akhir-SPBE-IF-2019/ta-spbe-textfinding/pull/5#discussion_r1142379894)

Comment (afifahfq) [reply textfinding_indikator10.py]: Thanks for the feedback! I've added new lines of code. Please give it another review if you've time.

**Review by akramab — APPROVED** [review]
LGTM

File: textfinding_indikator10.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        return ''
+
+    # TODO
+    # add setiap opd, etc
+
+    lvl2 = ['tugas']
+    res2 = txtreader(filename, 2, lvl2)
+
+    if (not res2):
+        return clean_text(list_final)
```
[the remark this answers is not in the document]
Comment (afifahfq) [reply textfinding_indikator10.py:82]: Thank you for pointing that out! I've fixed it according to the suggested changes. Please give it another review if you've time.

**Review by AwwalaK — APPROVED** [review]

**Pull request merged**